code401Challenges/src/main/java/code401Challenges/graph/Graph.java

```diff
@@ -47,9 +50,30 @@ public void addEdge(Node<T> node1, Node<T> node2, int weight) {
         }
     }
 
-
     // getSize() Returns the total number of nodes in the graph
     public int getSize() {
         return nodesInSet.size();
     }
+
+
+    // breadthFirstTraverse through graph
+    public HashSet<Node<T>> breadthFirstTraverse(Node<T> node) {
+        LinkedList<Node<T>> nodesToProcess = new LinkedList<>(); // will use as a queue
+        HashSet<Node<T>> seen = new HashSet<>();
+
+        nodesToProcess.add(node);
+
+        while(!nodesToProcess.isEmpty()) {
+            Node current = nodesToProcess.removeFirst();
+            seen.add(current);
+
+            HashSet neighbors = getNeighbors(current);
```

Comment:
The HashSet should also have a type declared. In this case, since it is holding generic nodes you would write: `HashSet<Node<T>> seen = new HashSet<>();`